### PR TITLE
[14.0][ADD] website_sale_infinite_scroll: new module

### PR DESCRIPTION
**Pull request by dessanhemrayev**
This module removes pagination in /shop page and replaces it with infinite scrolling of products.

### CODE REVIEW

File: website_sale_infinite_scroll/readme/CONFIGURE.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+To select different preload go to * General Settings > Website:
+ - Enabled "Website sale infinite scroll preloader"
+ - Select a new gif file
```
Comment (francesco-ooops) [inline website_sale_infinite_scroll/readme/CONFIGURE.rst]: Is this still valid?

File: website_sale_infinite_scroll/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+This module replaces the usual pagination in e-commerce
```
Comment (francesco-ooops) [inline website_sale_infinite_scroll/readme/DESCRIPTION.rst]: This module removes pagination in /shop page and replaces it with infinite scrolling of products.

**Review by francesco-ooops — COMMENTED** [review]
Add to USAGE

Go to /shop page > Customize > activate "Infinite Products Scroll"

File: website_sale_infinite_scroll/readme/CONFIGURE.rst (hunk 1 above)
Comment (dessanhemrayev) [reply website_sale_infinite_scroll/readme/CONFIGURE.rst]: yes)

Comment (francesco-ooops) [reply website_sale_infinite_scroll/readme/CONFIGURE.rst]: @dessanhemrayev can you attach a screenshot?

Comment (dessanhemrayev) [reply website_sale_infinite_scroll/readme/CONFIGURE.rst]: image
Okay, I fixed it

File: website_sale_infinite_scroll/controllers/main.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            category=category,
+            search=search,
+            ppg=self._check_website_sale_infinite_scroll(ppg),
+            **post
+        )
+        return request.render("website_sale.products", res.qcontext)
+
+    def _check_website_sale_infinite_scroll(self, ppg):
+        return (
+            21
```
Comment (GabbasovDinar) [inline website_sale_infinite_scroll/controllers/main.py]: why not website.shop_ppg?

Comment (GabbasovDinar) [reply website_sale_infinite_scroll/controllers/main.py]: or new field for infinity scroll

File: website_sale_infinite_scroll/controllers/main.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        )
+        if self._check_page(page, res.qcontext["pager"]["page_count"]):
+            return False
+        return request.render(
+            "website_sale_infinite_scroll.infinite_products",
+            res.qcontext,
+        )
+
+    def _check_page(self, page, page_count):
+        return page > page_count
```
Comment (GabbasovDinar) [inline website_sale_infinite_scroll/controllers/main.py]: I think it is unnecessary to put one logical operation into a separate method

File: website_sale_infinite_scroll/controllers/main.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        res = super(WebsiteSaleInfinityScroll, self).shop(
+            page=page,
+            category=category,
+            search=search,
+            ppg=self._check_website_sale_infinite_scroll(ppg),
+            **post
+        )
+        return request.render("website_sale.products", res.qcontext)
+
+    def _check_website_sale_infinite_scroll(self, ppg):
```
Comment (GabbasovDinar) [inline website_sale_infinite_scroll/controllers/main.py]: the name of the method does not correspond to the content

**Review by GabbasovDinar — CHANGES_REQUESTED** [review]

File: website_sale_infinite_scroll/controllers/main.py (hunk 3 above)
Comment (dessanhemrayev) [reply website_sale_infinite_scroll/controllers/main.py]: Thanks, I'll fix it

File: website_sale_infinite_scroll/controllers/main.py (hunk 5 above)
Comment (dessanhemrayev) [reply website_sale_infinite_scroll/controllers/main.py]: Thanks, I'll fix it

File: website_sale_infinite_scroll/readme/CONFIGURE.rst [hunk 6]
```diff
@@ -0,0 +1,2 @@
+To select different preload go to * General Settings > Website:
+ - "Website sale infinite scroll preloader" change preloader;
```
Comment (francesco-ooops) [inline website_sale_infinite_scroll/readme/CONFIGURE.rst]: To customize preloading icon, go to * General Settings > Website > Website sale infinite scroll preloader"

**Review by francesco-ooops — APPROVED** [review]
Functional ok!

**Review by GabbasovDinar — APPROVED** [review]

File: website_sale_infinite_scroll/models/product_template.py [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    @api.model
+    def create_dummy_products(self):
+        return self.create(
+            [
+                {"name": "product {num}".format(num=x), "website_published": True}
```
Comment (GabbasovDinar) [inline website_sale_infinite_scroll/models/product_template.py]: Maybe it makes sense to just reduce the number of products on the shop page?

**Review by GabbasovDinar — CHANGES_REQUESTED** [review]

File: website_sale_infinite_scroll/models/product_template.py (hunk 7 above)
Comment (dessanhemrayev) [reply website_sale_infinite_scroll/models/product_template.py]: Thank you, yes, everything is correct, but this is to test all the functionality, I will transfer it to the demo.

**Review by GabbasovDinar — APPROVED** [review]

**Comment by francesco-ooops** [comment]
@OCA/e-commerce-maintainers good for merge?

**Comment by francesco-ooops** [comment]
@OCA/e-commerce-maintainers what do you say?

**Comment by francesco-ooops** [comment]
@OCA/e-commerce-maintainers go? :)

**Comment by francesco-ooops** [comment]
@pedrobaeza can you merge based on existing reviews? 

thanks!

**Comment by pedrobaeza** [comment]
Well, I don't like too much this feature "as user", as this doesn't serve for linking pages to others, as there are no pages, so you can't "bookmark" anything in the list, but being a separate module, you can choose to not install it:

/ocabot merge nobump

**Comment by pedrobaeza** [comment]
Can you rebase on current main branch?

**Comment by francesco-ooops** [comment]
> Can you rebase on current main branch?

@dessanhemrayev

**Comment by pedrobaeza** [comment]
Please check the CI.

**Comment by HekkiMelody** [comment]
Failing CI is not due to this module. There's another problem that affects all PR on v14, I'm fixing it here: https://github.com/OCA/e-commerce/pull/839

**Pull request merged**